CesiumGltf/src/StructuralMetadataPropertyTableView.cpp

```diff
@@ -0,0 +1,494 @@
+    const gsl::span<const std::byte>& arrayOffsets,
+    const gsl::span<const std::byte>& stringOffsets,
+    size_t valueBufferSize,
+    PropertyComponentType stringOffsetType,
+    size_t propertyTableCount) noexcept {
+  const auto status = checkOffsetsBuffer<T>(
+      arrayOffsets,
+      stringOffsets.size(),
+      propertyTableCount,
+      false,
+      MetadataPropertyViewStatus::ErrorArrayOffsetsNotSorted,
+      MetadataPropertyViewStatus::ErrorArrayOffsetOutOfBounds);
+
+  if (status != MetadataPropertyViewStatus::Valid) {
+    return status;
+  }
+
+  const T* pValue = reinterpret_cast<const T*>(arrayOffsets.data());
+
+  switch (stringOffsetType) {
+  case PropertyComponentType::Uint8:
+    return checkOffsetsBuffer<uint8_t>(
+        stringOffsets,
+        valueBufferSize,
+        pValue[propertyTableCount] / sizeof(T),
+        false,
+        MetadataPropertyViewStatus::ErrorStringOffsetsNotSorted,
+        MetadataPropertyViewStatus::ErrorStringOffsetOutOfBounds);
+  case PropertyComponentType::Uint16:
+    return checkOffsetsBuffer<uint16_t>(
+        stringOffsets,
+        valueBufferSize,
+        pValue[propertyTableCount] / sizeof(T),
+        false,
+        MetadataPropertyViewStatus::ErrorStringOffsetsNotSorted,
+        MetadataPropertyViewStatus::ErrorStringOffsetOutOfBounds);
+  case PropertyComponentType::Uint32:
+    return checkOffsetsBuffer<uint32_t>(
+        stringOffsets,
+        valueBufferSize,
+        pValue[propertyTableCount] / sizeof(T),
+        false,
+        MetadataPropertyViewStatus::ErrorStringOffsetsNotSorted,
+        MetadataPropertyViewStatus::ErrorStringOffsetOutOfBounds);
+  case PropertyComponentType::Uint64:
+    return checkOffsetsBuffer<uint64_t>(
+        stringOffsets,
+        valueBufferSize,
+        pValue[propertyTableCount] / sizeof(T),
+        false,
+        MetadataPropertyViewStatus::ErrorStringOffsetsNotSorted,
+        MetadataPropertyViewStatus::ErrorStringOffsetOutOfBounds);
+  default:
+    return MetadataPropertyViewStatus::ErrorInvalidStringOffsetType;
+  }
+}
+
+MetadataPropertyTableView::MetadataPropertyTableView(
+    const Model* pModel,
+    const ExtensionExtStructuralMetadataPropertyTable* pPropertyTable)
+    : _pModel{pModel}, _pPropertyTable{pPropertyTable}, _pClass{nullptr} {
+  assert(pModel != nullptr && "pModel must not be nullptr");
+  assert(pPropertyTable != nullptr && "pPropertyTable must not be nullptr");
+
+  const ExtensionModelExtStructuralMetadata* pMetadata =
+      pModel->getExtension<ExtensionModelExtStructuralMetadata>();
+  assert(
+      pMetadata != nullptr &&
+      "Model must contain ExtensionModelExtStructuralMetadata to use "
+      "MetadataPropertyTableView");
+
+  const std::optional<ExtensionExtStructuralMetadataSchema>& schema =
+      pMetadata->schema;
+  assert(
+      schema != std::nullopt &&
+      "ExtensionModelExtStructuralMetadata must contain "
+      "Schema to use MetadataPropertyTableView");
```

Comment:
I added a `MetadataPropertyTableViewStatus` enum, similar to `MetadataPropertyViewStatus`. Now instead of `assert` statements, the constructor sets the status to an error value and exits early. If the property table view is invalid, then all of the property views it retrieves will be invalid as well.